File: Python3/1 - 200/137 Single Number II.py
# 64ms, 35.75%
class Solution:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        ones = twos = 0
        for n in nums:
            ones ^= ~twos & n
            twos ^= ~ones & n
        return ones

File: 1 - 200/137 Single Number II.cc
// 4ms, 100.0%
#include <bits/stdc++.h>
using namespace std;

static int x = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution {
public:
    int singleNumber(const vector<int> &nums) {
        int ones = 0, twos = 0;
        for (const auto n : nums) {
            ones ^= ~twos & n;
            twos ^= ~ones & n;
        }
        return ones;
    }
};
